src/main/java/bio/terra/service/policy/PolicyService.java

```diff
@@ -44,18 +46,29 @@ public PolicyService(PolicyApiService policyApiService) {
 
   // -- Policy Attribute Object Interface --
 
-  public static TpsPolicyInput getGroupConstraintPolicyInput(String groupName) {
-    TpsPolicyPair policyPair = new TpsPolicyPair().key(GROUP_CONSTRAINT_KEY_NAME).value(groupName);
-    return new TpsPolicyInput()
-        .namespace(POLICY_NAMESPACE)
-        .name(GROUP_CONSTRAINT_POLICY_NAME)
-        .addAdditionalDataItem(policyPair);
+  public static TpsPolicyInput getGroupConstraintPolicyInput(List<String> groupNames) {
+    var input = new TpsPolicyInput().namespace(POLICY_NAMESPACE).name(GROUP_CONSTRAINT_POLICY_NAME);
+    groupNames.forEach(
+        groupName -> {
+          input.addAdditionalDataItem(
+              new TpsPolicyPair().key(GROUP_CONSTRAINT_KEY_NAME).value(groupName));
```

Comment:
Since you're converting `groupNames` to `TpsPolicyPair` objects, could you do this as a map rather than forEach? Then you could also do this in the chained constructors.

Reply:
I'm not sure how to do that - we need to hit the "addAddtionalDataItem" method for each group name which I don't think would work within a map. Do you see how to do this?